router/router.go

```diff
@@ -60,7 +71,36 @@ func Create(db *database.GormDatabase, vInfo *model.VersionInfo, conf *config.Co
 		}
 	})
 
-	g.OPTIONS("/*any")
+	{
+		g.GET("/plugin", authentication.RequireClient(), pluginHandler.GetPlugins)
+		pluginRoute := g.Group("/plugin/", authentication.RequireClient())
+		{
+			pluginRoute.GET("/:id/config", pluginHandler.GetConfig)
+			pluginRoute.POST("/:id/config", pluginHandler.UpdateConfig)
+			pluginRoute.GET("/:id/config/schema", pluginHandler.GetConfigSchema)
+			pluginRoute.GET("/:id/display", pluginHandler.GetDisplay)
+			pluginRoute.POST("/:id/enable", pluginHandler.EnablePlugin)
+			pluginRoute.POST("/:id/disable", pluginHandler.DisablePlugin)
+		}
+		pluginHandler.CustomRouteMux = g.Group("/plugin/:id/custom/")
+	}
+
+	if pluginsDir := conf.PluginsDir; pluginsDir != "" {
+		pluginHandler.LoadPlugins(pluginsDir)
+	}
+	if err := pluginHandler.AddAllUsers(); err != nil {
+		panic(err)
+	}
+
+	if mode.IsDev() {
+		g.OPTIONS("/*any", func(ctx *gin.Context) {
```

Comment:
This was already done in line 63, why is it needed?

Reply:
The default handler to OPTIONS does not allow all HTTP Methods used in the API

Reply:
Really? I've removed this lines and made a quick curl request and the headers still were returned
```bash
λ curl -I -X OPTIONS http://localhost/
HTTP/1.1 200 OK
Access-Control-Allow-Headers: X-Gotify-Key,Authorization,Content-Type,Upgrade,Origin,Connection,Accept-Encoding,Accept-Language,Host
Access-Control-Allow-Methods: GET,POST,DELETE,OPTIONS,PUT
Access-Control-Allow-Origin: *
Content-Type: application/json
Date: Tue, 29 Jan 2019 19:47:28 GMT
Content-Length: 0
```

Do you have a config.yml? In there headers set in line 63 can be overriden.

Reply:
Okay, I got this. I copied the configuration from the example and that override the default behavior. I'll revert this. Thanks for your reminder:)